java/statestore-commit/src/main/java/sleeper/commit/StateStoreCommitter.java

```diff
@@ -120,4 +124,16 @@ private void apply(IngestAddFilesCommitRequest request) throws StateStoreExcepti
         }
     }
 
+    private void apply(CompactionJobIdAssignmentCommitRequest request) throws StateStoreException {
+        StateStore stateStore = stateStoreProvider.getByTableId(request.getTableId());
+        try {
+            stateStore.assignJobIds(request.getAssignJobIdRequests());
+        } catch (FileReferenceNotFoundException e) {
+            LOGGER.info("One or more files were not found in state store, skipping request", e);
+            return;
+        } catch (FileReferenceAssignedToJobException e) {
+            LOGGER.info("One or more files are already assigned to a job, skipping request", e);
+            return;
```

Comment:
I'm really sorry, I don't think we do need to handle these exceptions here. I thought it would retry the commit but it'll actually just go straight to the dead letter queue. That seems worthwhile, because it'll be easy to see something's gone wrong.